flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/writer/RecordWriter.java

```diff
@@ -113,11 +113,7 @@ public void broadcastEmit(T record) throws IOException, InterruptedException {
 	 * This is used to send LatencyMarks to a random target channel.
 	 */
 	public void randomEmit(T record) throws IOException, InterruptedException {
-		serializer.serializeRecord(record);
-
-		if (copyFromSerializerToTargetChannel(rng.nextInt(numChannels))) {
-			serializer.prune();
-		}
+		emit(record, new int[] { rng.nextInt(numChannels) });
```

Comment:
Wait a minute here. 

```
emit(record, new int[] { rng.nextInt(numChannels) });
```

@NicoK is an actual performance regression. Creating a single element int array (once per every record!) actually reduces the throughput by about 10%-20% (tested during one of the hackathons). Please revert to using `copyFromSerializerToTargetChannel(rng.nextInt(numChannels))`

Reply:
got it